docs/protocol.md

```diff
@@ -727,6 +728,16 @@ For example you may want to keep some in memory state, i.e. cache, aggregation.
 You can use read the `sk` and forward the call request to a specific process
 that has ownership for the shard key.
 
+### Transport Header `cr` -- Custom routing
+
+The custom routing request header must be set to a valid service name.
+
+When the custom routing header is set we will route to the service name
+in the `cr` header instead of the service name in the call request.
+
+It is expected that the service in the `cr` header is capable of
```

Comment:
If we want a request to hit a service identified by `cr` so that the app can then route it based on `arg1`, isn't that the same as sending the request to `arg1 = cr` and letting the app route based on `sk`? Is the uuid constraint on `sk` explicit or just expected by ringpop?